src/main/java/com/puppycrawl/tools/checkstyle/checks/blocks/RightCurlyCheck.java

```diff
@@ -559,13 +560,29 @@ private static Details getDetails(DetailAST ast) {
                 case TokenTypes.LITERAL_DO:
                     details = getDetailsForDoLoops(ast);
                     break;
+                case TokenTypes.LITERAL_SWITCH:
+                    details = getDetailsForSwitch(ast);
+                    break;
                 default:
                     details = getDetailsForOthers(ast);
                     break;
             }
             return details;
         }
 
+        /**
+         * Collect all the requirement to check violation for switch statements and expressions.
+         *
+         * @param ast Node which pointing to Token Switch case
+         * @return object containing all details to make a validation
+         */
+        private static Details getDetailsForSwitch(DetailAST ast) {
+            final DetailAST lcurly = ast.findFirstToken(TokenTypes.LCURLY);
+            final DetailAST rcurly = ast.getLastChild();
+            final DetailAST nextToken = getNextToken(ast);
```

Comment:
Can we avoid call to `getNextToken()`? If switch statements/expressions are the only token we need different logic than current `getNextToken()` provides, let's keep code to get next token confined to this method. It's likely that we can avoid a loop completely if we do this.

Reply:
> if we have to create the code will look like

No, this is too crazy, let's avoid such code. Please move your customized `getNextToken()` code into `getDetailsForSwitch` and let's see if pitest can help to simplify our approach. It looks like `getNextToken()` is used in a bunch of places, I would rather avoid complicating logic of this simple method if we can help it.

Reply:
Done